src/frontend/src/common/components/library/data_subview/index.tsx

```diff
@@ -301,6 +301,10 @@ const DataSubview: FunctionComponent<Props> = ({
       );
     }
 
+    const checkedSamples = compact(
+      checkedSampleIds.map((id) => data?.[id]) as Sample[]
+    );
+
```

Comment:
can you add a comment for what this logic is? I'm a little confused why you're using `compact` here